### PR TITLE
shipit-static-analysis: Do not check clang-tidy -* is present.

### LINKED ISSUE #1060: "-*" checker not found in available checkers
I've noticed this error in Sentry.

From https://hg.mozilla.org/mozilla-central/raw-file/tip/tools/clang-tidy/config.yaml:
```
clang_checkers:
 - name: -*
   publish: !!bool no
 - name: misc-forward-declaration-namespace
   # Name with clang tidy 6.0. We are currently using 5.0
   # - name: bugprone-forward-declaration-namespace
   publish: !!bool yes
 ...
```

### PR DESCRIPTION
**Pull request by La0**
[Sentry Error](https://sentry.prod.mozaws.net/operations/mozilla-releng-services/issues/3608196/)

Fixes #1060.

### CODE REVIEW

**Comment by marco-c** [comment]
This fixes https://github.com/mozilla-releng/services/issues/1060.

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py [hunk 1]
```diff
@@ -183,10 +183,17 @@ def list_missing_checks(self):
         else:
             logger.error('Firefox clang-tidy configuration {} should specify > 0 clang_checkers'.format(CONFIG_URL))
 
-        return [
+        # Special check -* is not listed by clang-tidy
+        required_checks = [
             check['name']
             for check in settings.clang_checkers
-            if not len(fnmatch.filter(available_checks, check['name'])) > 0
+            if check['name'] != '-*'
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Nit: You could just add this condition to the existing code, without defining a new `required_checks`

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py [hunk 2]
```diff
@@ -183,10 +183,17 @@ def list_missing_checks(self):
         else:
             logger.error('Firefox clang-tidy configuration {} should specify > 0 clang_checkers'.format(CONFIG_URL))
 
-        return [
+        # Special check -* is not listed by clang-tidy
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Nit: it'd be better to put this comment inline with the condition check

**Comment by marco-c** [comment]
How come this is intermittent? And why is the test @jankeromnes added not failing with this?

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py (hunk 1 above)
Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Good point, it's indeed simpler

**Comment by La0** [comment]
I don't think it's intermittent, but it's only triggered for patches with C/C++ code.

The unit test uses a [mock config file](https://github.com/mozilla-releng/services/blob/master/src/shipit_static_analysis/tests/mocks/config.yaml) that did not include `-*`. I just added it in the PR

**Pull request merged**